### PR TITLE
Dan/vaccination status datamart

### PR DESCRIPTION
**Pull request by DanPaseltiner**
This PR contain 3 additions to the scripts directory:

1. Tweaks to generate_covid_case_datamart.py based on scaling up to the final ELR CSV.
2. Addition of a script to generate a vaccination status datamart.
3. Addition of script that creates and covid breakthrough datamart by joining the case and vaccination datamarts.

### CODE REVIEW

**Comment by celestemespinoza** [comment]
Task linked: [CU-2892ttk Write a function that given a patient's immunization records and a date returns their covid vax status on the specified date.](https://app.clickup.com/t/2892ttk)

**Comment by celestemespinoza** [comment]
Task linked: [CU-2892un7 Write a script to generate a COVID case datamart including vax status](https://app.clickup.com/t/2892un7)

**Review by graham-skylight — COMMENTED** [review]
LGTM, but I'll leave it to someone better with pandas to give it an actual +1

File: src/scripts/generate_vaccination_status_datamart.py
Comment (spencer-skylight) [inline src/scripts/generate_vaccination_status_datamart.py:67]: Would this be slightly more efficient (and still correct) using `elif`?

File: src/scripts/generate_vaccination_status_datamart.py
Comment (spencer-skylight) [inline src/scripts/generate_vaccination_status_datamart.py:139]: I wonder if some of the assumptions we are making are valid.  Here we are assuming the dose chronologically after the current dose is the 2nd... Could there be an edge case where a person went a week later the first dose by mistake/based on bad info, and that one didn't count, but they went again in the true second dose range.  I don't think this logic would accept that right?  It's just something in the back of my mind - I do think this is good enough for our purposes right now.

File: src/scripts/assets/COVID_vaccination_codes.csv
Comment (DanPaseltiner) [reply src/scripts/assets/COVID_vaccination_codes.csv:1]: Agreed I'll work on that.

File: src/scripts/generate_vaccination_status_datamart.py
Comment (DanPaseltiner) [reply src/scripts/generate_vaccination_status_datamart.py:27]: That's a good call and something I was already considering. I'll split these up.

File: src/scripts/generate_vaccination_status_datamart.py
Comment (DanPaseltiner) [reply src/scripts/generate_vaccination_status_datamart.py:37]: I can test to confirm, but when one or both dates are null an NaT (Not a time-null object for times) should be returned.

File: src/scripts/generate_vaccination_status_datamart.py
Comment (DanPaseltiner) [reply src/scripts/generate_vaccination_status_datamart.py:67]: Yes definitely!

File: src/scripts/generate_vaccination_status_datamart.py
Comment (DanPaseltiner) [reply src/scripts/generate_vaccination_status_datamart.py:139]: This is exactly the sort of logic that would have to be driven, by the specific analysis question at hand. This approach applies the official definition of "fully vaccinated" very strictly. Obviously for real people in the real world life is a lot messier. Definitely something to discuss with someone who was interested in actually use some of this code in production.

File: src/scripts/generate_vaccination_status_datamart.py
Comment (DanPaseltiner) [reply src/scripts/generate_vaccination_status_datamart.py:67]: Done.

File: src/scripts/generate_vaccination_status_datamart.py
Comment (DanPaseltiner) [reply src/scripts/generate_vaccination_status_datamart.py:30]: Done

File: src/scripts/assets/COVID_vaccination_codes.csv
Comment (DanPaseltiner) [reply src/scripts/assets/COVID_vaccination_codes.csv:1]: On second thought I'm going to leave the separate for now. They are not both loaded in the same script and each script can run independently of the other. Doesn't quite make sense to require loading of unneeded data.

**Review by DanPaseltiner — COMMENTED** [review]
Final changes before merge. Addressing Spencer's comments.

**Review by spencer-skylight — APPROVED** [review]
Looks good.  Thanks Dan!

**Pull request merged**
